syntax = "proto3";

option go_package = "pb/game";
import "common/comp.proto";


// 计算属性，服务器端计算
message CalculatedAttributesPBComponent {
    uint64 attack_power = 1;   // 攻击力
    uint64 defense_power = 2;  // 防御力
}

// 衍生属性，服务器端计算
message DerivedAttributesPBComponent {
    uint64 max_health = 1;     // 最大生命值
}

// 角色状态，仅包含计算和衍生属性
message ActorStatusPBComponent {
    CalculatedAttributesPBComponent calculated_attributes = 1; // 计算属性
    DerivedAttributesPBComponent derived_attributes = 2;     // 衍生属性
}

